compiler/rustc_mir_build/src/builder/matches/mod.rs

```diff
@@ -1940,6 +1940,10 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
     /// in match tree lowering.
     fn merge_trivial_subcandidates(&mut self, candidate: &mut Candidate<'_, 'tcx>) {
         assert!(!candidate.subcandidates.is_empty());
+        if candidate.false_edge_start_block.is_none() {
+            candidate.false_edge_start_block = candidate.subcandidates[0].false_edge_start_block;
+        }
+
```

Comment:
I'd rather not do this here; the invariant is that candidates with subcandidates shouldn't have `pre_binding`/`otherwise`/`false_edge_start` blocks, so we should only assign that after merge. This should be assigned in `remove_never_subcandidates` instead I think.

Reply:
Thanks. I modified it in https://github.com/rust-lang/rust/pull/135409/commits/481ed2d9319a28b770da60f4567e55f5cc053835.